winton_kafka_streams/state/change_logging_key_value_store.py

```diff
@@ -0,0 +1,29 @@
+from .store_change_logger import StoreChangeLogger
+
+class ChangeLoggingKeyValueStore:
+    def __init__(self, name, inner):
+        self.inner = inner(name)
+
+
+    def initialise(self, context, root):
+        self.inner.initialise(context, root)
+        self.change_logger = StoreChangeLogger(self.inner.name, context)
+
+
+    def __setitem__(self, key, value):
+        self.inner.__setitem__(key, value)
+        self.change_logger.log_change(key, value)
+
+
+    def __getitem__(self, key):
+        return self.inner.__getitem__(key)
+
+
+    def get(self, key, default=None):
+        return self.inner.get(key, default)
+
+
+    def __delitem_(self, key):
```

Comment:
Missing underscore on end of method name.

Reply:
Good spot! Will fix.

Reply:
Might be worth testing it works as a dict: to test the special methods.